Kernel/Modules.cmake

```diff
@@ -0,0 +1,162 @@
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS "${CMAKE_SOURCE_DIR}/Kernel/config.ini")
+
+macro( readSettingFile KEY DEFAULT_RESULT STRING_RESULT_OUT)
```

Comment:
It's unfortunate that this reads the file from 0 for each variable we want to look up.

I think a simpler design if we want to keep the separate ini file might be something like

```cmake
file(READ "${CMAKE_SOURCE_DIR}/Kernel/config.ini" contents )
string(REGEX REPLACE ";" "\\\\;" contents "${contents}")
string(REGEX REPLACE "\n" ";" contents "${contents}")
foreach(line IN LISTS contents)
    string(REPLACE "=" ";" line_list line)
    list(GET line_list 0 var)
    list(GET line_list 1 value)
    set("${var}" "${value}" CACHE BOOL FORCE) # force the changes from the file to take effect
endforeach()
```

If you did this, then you can just check 
```
if (COMPILE_RAMDISK)
   # add ramdisk cpp and define
endif()
```